DllImportGenerator/DllImportGenerator/DllImportStub.cs

```diff
@@ -123,76 +113,56 @@ public class GeneratedDllImportData
                 stubTypeNamespace = method.ContainingNamespace.ToString();
             }
 
-            // Determine type
-            var stubContainingTypes = new List<string>();
+            // Determine containing type(s)
+            var containingTypes = new List<TypeDeclarationSyntax>();
             INamedTypeSymbol currType = method.ContainingType;
             while (!(currType is null))
             {
                 var visibility = currType.DeclaredAccessibility switch
                 {
-                    Accessibility.Public => "public",
-                    Accessibility.Private => "private",
-                    Accessibility.Protected => "protected",
-                    Accessibility.Internal => "internal",
+                    Accessibility.Public => SyntaxKind.PublicKeyword,
+                    Accessibility.Private => SyntaxKind.PrivateKeyword,
+                    Accessibility.Protected => SyntaxKind.ProtectedKeyword,
+                    Accessibility.Internal => SyntaxKind.InternalKeyword,
                     _ => throw new NotSupportedException(), // [TODO] Proper error message
                 };
 
-                var typeKeyword = currType.TypeKind switch
+                TypeDeclarationSyntax typeDecl = currType.TypeKind switch
                 {
-                    TypeKind.Class => "class",
-                    TypeKind.Struct => "struct",
+                    TypeKind.Class => ClassDeclaration(currType.Name),
+                    TypeKind.Struct => StructDeclaration(currType.Name),
                     _ => throw new NotSupportedException(), // [TODO] Proper error message
                 };
 
-                stubContainingTypes.Add($"{visibility} partial {typeKeyword} {currType.Name}");
+                typeDecl = typeDecl.AddModifiers(
+                    Token(visibility),
+                    Token(SyntaxKind.PartialKeyword));
+                containingTypes.Add(typeDecl);
+
                 currType = currType.ContainingType;
             }
 
-            // Flip the order to that of how to declare the types
-            stubContainingTypes.Reverse();
-
-            // Determine parameter types
+            // Determine parameter and return types
             var paramsTypeInfo = new List<TypePositionInfo>();
-            foreach (var paramSymbol in method.Parameters)
+            foreach (var param in method.Parameters)
             {
-                paramsTypeInfo.Add(TypePositionInfo.CreateForParameter(paramSymbol));
+                paramsTypeInfo.Add(TypePositionInfo.CreateForParameter(param, compilation));
             }
 
-            var retTypeInfo = TypePositionInfo.CreateForType(method.ReturnType, method.GetReturnTypeAttributes());
+            var retTypeInfo = TypePositionInfo.CreateForType(method.ReturnType, method.GetReturnTypeAttributes(), compilation);
 
+            // Generate stub code
```

Comment:
Are we no longer interested in the `GENERATE_FORWARDER`? I liked this for testing purposes. Not sure it is still valuable though.

Reply:
I put back a version of this - it still goes through this generation, but we always use the `Forwarder` implementation of `IMarshallingGenerator`. If it gets more complicated to keep, we can remove it.